R/utils.R

```diff
@@ -1,17 +1,18 @@
-recodeValues <- function(values, desiredValues, dataType) {
-  if (is.null(desiredValues)) return(values)
+recodeValues <- function(values, desiredValues) {
+  if (!length(desiredValues)) return(values)
+  desiredLabels <- unlist(lapply(desiredValues, function(x) {x@binLabel}))
```

Comment:
do desired values have to have a binLabel? I might be thinking of a different class, but i thought binLabels were optional? What happens if there is no bin label?

Reply:
all Bins have a binLabel, everything else is optional technically. but there is also validation that requires if a single binStart or binEnd is provided in a BinList, then they all must be.